src/main/groovy/com/github/lkishalmi/gradle/gatling/GatlingExtension.groovy

```diff
@@ -28,4 +31,27 @@ class GatlingExtension {
     def simulations = {
         include "**/*Simulation.scala"
     }
+
+    private final boolean isGatlingLayout
+
+    GatlingExtension(Project project) {
+        this.isGatlingLayout = project.file("src/gatling/simulations").exists() && project.file("src/gatling/data").exists() && project.file("src/gatling/bodies").exists()
+    }
+
+    String simulationsDir() {
+        "src/gatling/${ isGatlingLayout ? "simulations" : "scala" }"
+    }
+
```

Comment:
Why can't dataDir() and bodiesDir() return ```resources/bodies``` and ```resources/data``` respectively and use them in gatlingArgs and in sourceSets.garling.resources.srcDirs as well.